spyder/plugins/projects/api.py

```diff
@@ -94,38 +94,67 @@ def get_recent_files(self):
 
         return list(OrderedDict.fromkeys(recent_files))
 
-    def get_plugin(self, plugin_name):
-        """
-        Return a plugin by unique `plugin_name`.
-        """
-        return self.projects_plugin.get_plugin(plugin_name)
-
     # --- API
     # ------------------------------------------------------------------------
     @staticmethod
-    def get_name(self):
+    def get_name():
         """
         Provide a human readable version of NAME.
         """
         raise NotImplementedError("Must implement a `get_name` method!")
 
+    @staticmethod
+    def validate_name(path, name):
+        """
+        Validate the name of the project.
+
+        Returns
+        -------
+        tuple
+            The first item (bool) indicates if the name was validated
+            successfully, and the second item (str) indicates the error
+            message, if any.
+        """
+        return True, ""
+
     def create_project(self):
         """
         Create a project and do any additional setup for this project type.
+
+        Returns
+        -------
+        tuple
+            The first item (bool) indicates if the project was created
+            successfully, and the second item (str) indicates the error
+            message, if any.
         """
-        raise NotImplementedError("Must implement a `create_project` method!")
+        return None, "A ProjectType must define a `create_project` method!"

```

Comment:
This doesn't correspond to the docstring you wrote above because you're not returning a bool but `None`.

Reply:
Fixed